README.md

```diff
@@ -1,4 +1,154 @@
-# `hello-juju-charm`
+# Hello, Juju! Charm
 
-This repository contains the source code for introductory charms
-for new Juju users.
+This [charm](https://charmhub.io/hello-juju) is a demonstration machine charm
+written with the [Charmed Operator Framework](https://github.com/canonical/operator).
+
+It deploys a simple Python [Flask](https://flask.palletsprojects.com/en/2.0.x/)
+[web application](https://github.com/juju/hello-juju). The application itself is
+simple, and counts the number of times the root URL has been requested and stores the
+count in a database. With no database relation defined, this information is stored in
+a sqlite database.
+
+This charm also supports a [relation](https://juju.is/docs/sdk/relations) to the
+[PostgreSQL charm](https://charmhub.io/postgresql). When this relation is established
+the Flask application is automatically configured to use the PostgreSQL database
+to store its count.
+
+## Quickstart
+
+Assuming you already have Juju installed and bootstrapped on a cloud, such as LXD:
+
+```bash
+# Create a model
+$ juju add-model dev
+# Deploy the charm
+$ juju deploy hello-juju
+# Wait for the deployment to complete
+$ juju status
+Model  Controller  Cloud/Region         Version  SLA          Timestamp
+dev    lxd         localhost/localhost  2.9.1    unsupported  14:46:51+01:00
+
+App         Version  Status  Scale  Charm       Store  Channel  Rev  OS      Message
+hello-juju           active      1  hello-juju  local             0  ubuntu
+
+Unit           Workload  Agent  Machine  Public address  Ports   Message
+hello-juju/0*  active    idle   0        10.14.25.117    80/tcp
+
+Machine  State    DNS           Inst id        Series  AZ  Message
+0        started  10.14.25.117  juju-9f46aa-0  focal       Running
+```
+
+You should be able to visit [http://10.14.25.117](http://10.14.25.117)
```

Comment:
Maybe don't make this a link, and clarify that it's not literally `10.14.25.117`, but the "public address listed above" or something. Similar below.

Reply:
Fixed in next commit, thanks! :)